1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,181 @@
+---
+RFC: Update $ErrorView with Single String Error Messages
+Author: Jason Helmick
+Status: Draft
+Version: 0.0.0
+Area: PowerShell
+Comments Due: 
+---
+
+# Update $ErrorView with Single String Error Messages
+
+When an error occurs in PowerShell, the customers on-screen error message experience currently
+provides a level of detail that obscures the inner exception message from being read by
+new or occasional PowerShell users.
+To improve both comprehension and the troubleshooting experience, a single string of text
+containing the error will be displayed.  A cmdlet will provide the full error details when desired.
+
+## Motivation
+
+The on-screen experience, when receiving an error message,
+is controlled through the views NormalView (the default) and CategoryView. These are user selectable though the preference variable $ErrorView.
+The addition of a “ConciseView” as default that contains only specific error
+information producing a single line on screen and in logs will improve customer success.
+
+Comprehensive error information is available to the customer through
+accessing the error object. For new and occasional customers,
+a cmdlet “Resolve-ErrorRecord” will simplify the process of getting detailed error information, improving customers success.
+
+## Specification
+
+The proposal is to add a new single-line default error message view called ConciseView
+and add a new cmdlet (Resolve-ErrorRecord) to provide detailed error information.
+
+__Key Design Considerations__
+
+1. To reduce confusion and improve debugging success,
+error messages by default should produce a single line of text, including the word “ERROR:” to make consistent with Verbose and Warning messages. This is added to $errorView as view named "ConciseView".
+
+- $ErrorView should contain these views
+
+    + NormalView
+    + CategoryView
+    + ConciseView
+
```

Comment:
IIRC in the implementation behind this, the views right now are `CategoryView` if you enter it without typos, and `NormalView` for any other view (i.e. you could set the default view today to "foo", and you'd end up with `NormalView`). With that in mind, the name `NormalView` could be changed because it's the "else" clause. Why does that matter? Because a better name for `NormalView` would be `DetailedView`, and I think a better name for `ConsiseView` would be `SimpleView`. As long as `DetailedView` is the view that is used when the name is anything other than `CategoryView` and `SimpleView`, it should just work.

Reply:
Now that I'm back at my office, I just checked and confirmed that the only reference to `NormalView` in the code base is when assigning the default value to the `$ErrorView` variable [here](https://github.com/PowerShell/PowerShell/blob/2ed9abc99b9ea9caf13b09b1b64099a86405794d/src/System.Management.Automation/engine/InitialSessionState.cs#L4414-L4418). In the F&O layer, PowerShell simply checks if [`$ErrorView -ne 'CategoryView'`](https://github.com/PowerShell/PowerShell/blob/482c64b35f8d1e167498467436c6e81a9325d940/src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs#L746) or [`$ErrorView -eq 'CategoryView'`](https://github.com/PowerShell/PowerShell/blob/482c64b35f8d1e167498467436c6e81a9325d940/src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs#L844), so the `NormalView` name could be changed without breaking anything

Reply:
From my own experience, I would actually like `DetailedView` to automatically show all the inner exceptions which is usually where the real root cause is.  So the current view could still be called `NormalView` and add a new `DetailedView` that traverses the inner exceptions.  Agree with calling it `SimpleView` instead of `ConciseView`.